src/Listeners/QueueJobSubscriber.php

```diff
@@ -0,0 +1,161 @@
+
+        if (in_array(ShouldBeTraced::class, class_implements($jobInstance))) {
+            $jobInput = $this->retrieveQueueJobInput($jobInstance);
+
+            $span = Trace::startSpan(class_basename($job->resolveName()), $jobInput->first(function ($attr) {
+                return $attr instanceof SpanContext;
+            }));
+
+            $span->tag('type', 'queue');
+            $span->tag('connection_name', $event->connectionName);
+            $span->tag('queue_name', $jobInstance->queue);
+            $span->tag('job_input', json_encode($this->normalizeQueueJobInputForLogging($jobInput)));
+
+            $this->app->instance('tracing.queue.span', $span);
+        }
+    }
+
+    /**
+     * @param JobProcessed|JobFailed $event
+     */
+    protected function closeQueueJobSpan($event)
+    {
+        if ($this->app->has('tracing.queue.span')) {
+            if ($event instanceof JobFailed) {
+                $this->app->get('tracing.queue.span')->tag('error', 'true');
+            }
+
+            $this->app->get('tracing.queue.span')->finish();
+
+            // If it's a sync driver, the main process will flush, we don't want to do that prematurely
+            if ($event->connectionName != 'sync') {
+                Trace::flush();
+            }
+        }
+    }
+
+    /**
+     * @param mixed $jobInstance
+     * @return Collection
+     */
+    protected function retrieveQueueJobInput($jobInstance): Collection
+    {
+        $class = new ReflectionClass($jobInstance);
+        $constructor = $class->getConstructor();
+
+        if ($constructor) {
+            return collect($constructor->getParameters())
+                ->mapWithKeys(function (ReflectionParameter $param) use ($class, $jobInstance) {
+                    $prop = $class->getProperty($param->name);
+
+                    $prop->setAccessible(true);
+
+                    return [$param->name => $prop->getValue($jobInstance)];
+                });
+        } else {
+            return new Collection();
+        }
+    }
+
+    /**
+     * @param  Collection  $jobInput
+     * @return array
+     */
+    protected function normalizeQueueJobInputForLogging(Collection $jobInput): array
+    {
+        return $jobInput
+            ->reject(function ($attr) {
+                return $attr instanceof SpanContext;
+            })
+            ->map(function ($attr) {
+                if (is_scalar($attr) || is_array($attr)) {
+                    return $attr;
+                }
+
+                if (is_object($attr) && $attr instanceof Arrayable) {
+                    return $attr->toArray();
+                }
+
+                if (is_object($attr) && $attr instanceof Jsonable) {
+                    return $attr->toJson();
+                }
+
+                if (is_object($attr) && $attr instanceof JsonSerializable) {
+                    return $attr->jsonSerialize();
+                }
+
```

Comment:
how about an eventual `if(is_object)` that would simply `serialize`? the possibility of receiving an object of "unknown type"

Reply:
Hm, currently we simply print public properties for object's of "unknown type". I am not fond of `serialize` to be honest because it's not human readable.